From d5dd157a7ae4a1201a48b5fe187e833eab27a205 Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Mon, 21 Nov 2022 14:43:26 +0530
Subject: [PATCH] ath12k : delete the timer rx_replenish_retry during rmmod

The timer rx_replenish_retry is not deleted during rmmod.
Delete the timer in ath12k_core_free.

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath12k/core.c b/drivers/net/wireless/ath/ath12k/core.c
index 6cfe955..d3395b5 100644
--- a/drivers/net/wireless/ath/ath12k/core.c
+++ b/drivers/net/wireless/ath/ath12k/core.c
@@ -1656,6 +1656,7 @@ void ath12k_core_deinit(struct ath12k_base *ab)
 
 void ath12k_core_free(struct ath12k_base *ab)
 {
+	del_timer_sync(&ab->rx_replenish_retry);
         flush_workqueue(ab->workqueue_aux);
         destroy_workqueue(ab->workqueue_aux);
 
-- 
2.7.4

